### DISCUSSION TITLE
How to set the taskbar icon on Linux?

### DISCUSSION THREAD

**Discussion by ghost** [coderedart/egui_overlay#31] [Q&A] [answered]
`cargo run --example basic`
Then it'll display the default x11 icon on Kubuntu 22.04.
I was able to set the exe icon for Windows 10 using the winres crate.
https://github.com/emilk/egui/discussions/3971
https://github.com/emilk/egui/discussions/2026

Is this function? How to do it when using egui_overlay?
18 pub fn load_icon() -> eframe::IconData
https://github.com/MrTanoshii/rusty-autoclicker/blob/main/src/utils.rs
or
https://github.com/emilk/egui/issues/3370
ViewportCommand::Icon with ctx.send_viewport_cmd
https://docs.rs/egui/latest/egui/viewport/enum.ViewportCommand.html
Icon(Option<Arc<IconData>>)
or
https://www.glfw.org/docs/latest/window_guide.html#window_icon
I got stuck in an icon data type error. And which one should I use?

**Comment by coderedart** [comment]
https://docs.rs/glfw-passthrough/latest/glfw_passthrough/struct.Window.html#method.set_icon_from_pixels

did you try using glfw directly? we dont support viewport commands as multiple windows with overlay is a little complex.

glfw eample https://github.com/PistonDevelopers/glfw-rs/blob/b8845df2142c868557b1dc6687c0b71540203627/examples/window_icon.rs#L42

**Comment by ghost** [comment]
```
if let DynamicImage::ImageRgba8(icon) = open_image("assets/icon.png").unwrap() {
            glfw_backend.window.set_icon_from_pixels(vec![
                resize(&icon, 16, 16, Nearest),
                resize(&icon, 32, 32, Nearest),
                resize(&icon, 48, 48, Nearest),
            ]);
        }
```
error[E0308]: mismatched types
note: expected struct `PixelImage`
        found struct `ImageBuffer<image::Rgba<u8>, std::vec::Vec<u8>>`
I couldn't solve this.

**Reply by coderedart** [reply, marked as answer]
ah, because image feature is not enabled by default.  you also seem to be using `set_icon_from_pixels` which is different from `set_icon`.
add the following dependency to your project, this should enable image feature for glfw crate across your entire dependency tree (including egui_overlay.
```toml
[dependencies]
egui_window_glfw_passthrough = { workspace = true, features = ["image"] } # or glfw-passthrough crate. any of them will work
```

Anyway, I added an example to the repo now which shows how to use `set_icon_from_pixels`. You don't really need to enable any new features. Just load your image, convert it into rgba8 bytes and set it as the icon.
https://github.com/coderedart/egui_overlay/blob/master/examples/icon/src/main.rs#L36

**Reply by ghost** [reply]
It works fine. Thanks a lot.
